crypto/test/bls/indy_crypto/test_bls_crypto_indy_crypto.py

```diff
@@ -47,38 +49,43 @@ def test_default_params(default_params):
     assert isinstance(g, str)
 
 
-def test_generate_keys_no_seed(default_params):
-    sk, pk = BlsCryptoSignerIndyCrypto.generate_keys(default_params)
+def test_generate_keys_no_seed(default_params, bls_verifier):
+    sk, pk, key_proof = BlsCryptoSignerIndyCrypto.generate_keys(default_params)
     assert sk
     assert isinstance(sk, str)
     assert pk
     assert isinstance(pk, str)
+    assert key_proof
+    assert isinstance(key_proof, str)
     assert sk != pk
+    assert bls_verifier.verify_key_proof_of_possession(key_proof, pk)
```

Comment:
Please add more tests for verification of PoP. Please add tests for negative cases (where verification fails).